internal/component/prometheus/remotewrite/types.go

```diff
@@ -293,9 +348,64 @@ func (a *AzureADConfig) Validate() error {
 		return fmt.Errorf("must provide a cloud in the Azure AD config")
 	}
 
-	_, err := uuid.Parse(a.ManagedIdentity.ClientID)
-	if err != nil {
-		return fmt.Errorf("the provided Azure Managed Identity client_id provided is invalid")
+	if a.ManagedIdentity == nil && a.OAuth == nil && a.SDK == nil {
+		return fmt.Errorf("must provide an Azure Managed Identity, Azure OAuth or Azure SDK in the Azure AD config")
+	}
+
+	if a.ManagedIdentity != nil && a.OAuth != nil {
+		return fmt.Errorf("cannot provide both Azure Managed Identity and Azure OAuth in the Azure AD config")
+	}
+
+	if a.ManagedIdentity != nil && a.SDK != nil {
+		return fmt.Errorf("cannot provide both Azure Managed Identity and Azure SDK in the Azure AD config")
+	}
+
+	if a.OAuth != nil && a.SDK != nil {
+		return fmt.Errorf("cannot provide both Azure OAuth and Azure SDK in the Azure AD config")
+	}
+
+	if a.ManagedIdentity != nil {
+		if a.ManagedIdentity.ClientID == "" {
+			return fmt.Errorf("must provide an Azure Managed Identity client_id in the Azure AD config")
+		}
+
+		_, err := uuid.Parse(a.ManagedIdentity.ClientID)
+		if err != nil {
+			return fmt.Errorf("the provided Azure Managed Identity client_id is invalid")
+		}
+	}
+
+	if a.OAuth != nil {
+		if a.OAuth.ClientID == "" {
+			return fmt.Errorf("must provide an Azure OAuth client_id in the Azure AD config")
+		}
+		if a.OAuth.ClientSecret == "" {
+			return fmt.Errorf("must provide an Azure OAuth client_secret in the Azure AD config")
+		}
+		if a.OAuth.TenantID == "" {
+			return fmt.Errorf("must provide an Azure OAuth tenant_id in the Azure AD config")
+		}
+
+		var err error
+		_, err = uuid.Parse(a.OAuth.ClientID)
```

Comment:
Is this specific to Azure? Not sure in generic oauth this is true.

Reply:
I'm not sure, but this is how [Prometheus checks Azure client IDs](https://github.com/prometheus/prometheus/blob/8cd9069cf192c88bfd8890280c78a6d583bcc694/storage/remote/azuread/azuread.go#L154). This code is copied from there.